docs/protocol_overview.md

```diff
@@ -0,0 +1,245 @@
+
+The condition 1 is actually turned out to be a tough challenge in the later sections, so keep in mind it.
+
+### On-demand block production
+
+Rarely-online nodes means rarely-progressed blockchain.
+(note that in any BFT algorithm, at least 2/3 of nodes must have participated in the consensus process to produce a block)
+
+This will inevitably compromise the latency or throughput of the blockchain, but it really doesn't matter because
+**Simperby is NOT a general contract platform for serving public users,**
+**but a governance platform for a permissioned set of members**.
+
+Simperby blockchain includes only governance-approved state transitions (there are very [few exceptions](#consensus-leader)).
+Every governance agenda must be approved by a majority vote of the members so it will take a time anyway (days or even weeks).
+Considering that, naturally, Simperby's performance will be bounded by the governance process, not the consensus slowed by lightweight node operation.
+
+### Multichain DAO
+
+One of the notable use cases of Simperby is that it can establish a **muli-chain DAO**. (See [this](./multichain_dao.md) for more details)
+it's super-easy to expand the organization to other existing chains (so called 'colony chains') using
+the light client of the Simperby consensus.
+Since the light client is uploaded as a contract to other chains,
+any block progress of the Simperby consensus will require an additional
+transaction to update and store the Merkle root of it.
+This might not be cheap, because **the organization will have to pay for the gas cost for every colony chains, for every Simperby block**.
+That is another reason that on-demand block production is considered to be reasonable.
+
+## Governance
+
+Simperby's governance is realized by the P2P voting.
+
+- Any member can propose an agenda, and it will be propagated over the P2P network.
+- Any member also can cast a vote to the agenda if they prefer it. The vote will be propagated as well.
+- Any agenda that has over 50% votes is **eligible** to be included in a block.
+- A block can be **valid only if it includes an eligible agenda**. In other words, empty blocks are not allowed.
+- To decide which eligible agenda to finalize is the role of consensus, not the governance.
+- The consensus is a BFT algorithm based on a leader-and-round style.
+
+### What can be an agenda?
+
+- Transaction is only the unit of a **state transition**, not an individual item signed and broadcasted as in the case of a conventional blockchain.
+- An 'agenda' is defined as `(Proposer, Height, [Transaction])`.
+- Only a **single agenda** can be included in a block. This is for preventing complicated dependency problem.
+  - If multiple agendas are allowed, each agenda must be dependency-free; Otherwise, voters can't be sure how each agenda is ordered, or even whether it is included in the finalized block. This will cause a severe restriction of the possible agenda items.
```

Comment:
```suggestion
  - If multiple agendas are allowed, each agenda must be independent; Otherwise, voters can't be sure how each agenda is ordered, or even whether it is included in the finalized block. This will cause a severe restriction of the possible agenda items.
```